### PR TITLE
Uses one test-requirements file for the project

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

One test-requirements file is added at the root of the project to be used by tox for all charms.

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -13,4 +13,5 @@ pytest==7.3.1
 pytest-operator==0.26.0
 types-PyYAML==6.0.12.9
 types-setuptools==67.6.0.7
-types-toml==0.10.8.6
\ No newline at end of file
+types-toml==0.10.8.6
+juju==3.0.4
```
Comment (gruyaume) [inline test-requirements.txt:17]: Aren't we using juju 2.9?

Comment (saltiyazan) [reply test-requirements.txt:17]: 3.0.4 is the latest version of `python-libjuju` that was tested with `juju 2.0+`
https://github.com/juju/python-libjuju/tree/3.0.4

**Review by ghislainbourgeois — APPROVED** [review]
The description of the PR does not seem to match the title or the content. Otherwise, LGTM

**Comment by saltiyazan** [comment]
> The description of the PR does not seem to match the title or the content. Otherwise, LGTM

I updated the description :)

**Review by Gmerold — APPROVED** [review]

**Pull request merged**